### ISSUE TITLE
[bug] 

### ISSUE THREAD

**Issue by melba-lopez** [slsa-framework/slsa-github-generator#547]
**Describe the bug**
Improve repository's OpenSSF Scorecard score (currently at 7.1)

**To Reproduce**
`docker run -e GITHUB_AUTH_TOKEN gcr.io/openssf/scorecard:stable --show-details --repo=https://github.com/slsa-framework/slsa-github-generator --format=json > scorecard_slsa-framework_slsa-github-generator.json`

**Expected behavior**

- Branch Protections could be improved
- CII-Best-Practices Badge could be obtained
- Project should be Fuzzed
- Security Policy should be created
- Token Permissions should follow principle of least priveledge

**Screenshots**
![image](https://user-images.githubusercontent.com/101211710/179085462-1b9e1eef-4a10-4bbc-89e8-d7b2c79c0d91.png)
![image](https://user-images.githubusercontent.com/101211710/179085517-21910b84-a436-4cc7-92ee-39ad5e09787e.png)
![image](https://user-images.githubusercontent.com/101211710/179086284-7c96bd7c-a6c3-49f1-ae82-0a6d5aaedb07.png)
![image](https://user-images.githubusercontent.com/101211710/179086133-4b000ab8-a769-4654-a57d-1f984a4b07f0.png)
![image](https://user-images.githubusercontent.com/101211710/179085764-dd56ce86-4883-41c8-a5c8-b1123680e02e.png)

**Additional context**
Attempted to upload the JSON file, but github does not allow me to.  Related to recommendation of securing our repos: https://github.com/slsa-framework/slsa/issues/424

**Comment by laurentsimon** [comment]
You know you can send us some PR and get rewarded at sos.dev, right?

**Comment by melba-lopez** [comment]
I thought PRs were mostly meant for doc/code changes??  I know I am unable to make these changes myself since I don't have repo access.  So not sure how that would work?

**Comment by laurentsimon** [comment]
The permission changes could be changes you propose via a PR. If you identify some critical projects and get them to accept your PR, you can use sos.dev to be rewarded

**Comment by laurentsimon** [comment]
I thought we already set the permissions as read-only in all our workflows. If we don't for some, please feel free to send a PR

**Comment by jspeed-meyers** [comment]
Huh, `scorecard` says, as of 10/19/22 for this repo, `0/10` for `Token-Permissions` and that `non read-only tokens detected`. Let me do a little digging.

**Comment by jspeed-meyers** [comment]
Could it be this line: https://github.com/slsa-framework/slsa-github-generator/blob/61d71969e88ad66f920aa17d86d822f428aa49cf/.github/workflows/generator_container_slsa3.yml#L88-L91

Seems like the keyless signing requires non-read permissions, so I'm not sure there's any way to get around this `0/10` scorecard score. If I am misunderstanding, please correct me and I'm glad to put in a PR.

**Comment by laurentsimon** [comment]
You're right. Scorecard scoring is going to change in the next release to address this. See https://github.com/ossf/scorecard/issues/2338

**Comment by ianlewis** [comment]
We have a score of 9 now but I think we can close once we've completed the following tasks that were written in the description.

- [x] **Branch Protections could be improved**: we have branch protections in place. scorecard can't detect it because it needs admin access. Using a PAT with scorecard is not recommended so this is what it is for now.
- [x] **CII-Best-Practices Badge could be obtained**: we have a badge and are working towards it. Tracked on #886 
Project should be Fuzzed
- [ ] **Security Policy should be created**: tracked by #1262 
- [x] **Token Permissions should follow principle of least priveledge**: we've made some changes and I think we are using least-privilege now.

**Comment by ianlewis** [comment]
It seems that scorecard also dings us hard for using `packages: write`. It's giving us a zero for the 'Token-Permissions' check.

**Comment by laurentsimon** [comment]
This should go away with one of the future releases of Scorecard
